File: frontend/src/app/models/purchaseDto.ts
import { Address } from './order';
import { OrderItem } from './orderItem';

export type PurchaseDto = {
  customer: { firstName: string; lastName: string; email: string };
  shippingAddress?: Address;
  billingAddress?: Address;
  order?: {
    totalQuantity: number;
    totalPrice?: number;
    status?: string;
  };
  orderItems?: OrderItem[];
};

File: backend/src/main/java/com/music/shop/dto/PurchaseDto.java
package com.music.shop.dto;

import java.util.Set;

import com.music.shop.entity.Address;
import com.music.shop.entity.Customer;
import com.music.shop.entity.Order;
import com.music.shop.entity.OrderItem;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PurchaseDto {
    private Customer customer;

    private Address shippingAddress;

    private Address billingAddress;

    private Order order;

    private Set<OrderItem> orderItems;
}
